client/my-sites/upgrades/trials/start-trial/free-trial-form.jsx

```diff
@@ -0,0 +1,86 @@
+/**
+ * External dependencies
+ */
+import React from 'react';
+import { connect } from 'react-redux';
+import Dispatcher from 'dispatcher';
+
+/**
+ * Internal dependencies
+ */
+import analytics from 'analytics';
+import { cartItems } from 'lib/cart-values';
+import TransactionStepsMixin from 'my-sites/upgrades/checkout/transaction-steps-mixin';
+import FreeTrialConfirmationBox from './free-trial-confirmation-box';
+import { clearSitePlans, fetchSitePlans } from 'state/sites/plans/actions';
+import { getPlansBySiteId } from 'state/sites/plans/selectors';
+
+const FreeTrialForm = React.createClass( {
+	mixins: [ TransactionStepsMixin ],
+
+	componentDidMount: function() {
+		this.props.fetchSitePlans( this.props.selectedSite.ID );
+	},
+
+	componentWillReceiveProps: function( nextProps ) {
+		if ( this.props.selectedSite.ID !== nextProps.selectedSite.ID ) {
+			this.props.fetchSitePlans( nextProps.selectedSite.ID );
+		}
+	},
+
+	handleSubmit: function( event ) {
+		analytics.ga.recordEvent( 'Upgrades', 'Submitted Free Trial Form' );
+
+		// `submitTransaction` comes from the `TransactionStepsMixin`
+		this.submitTransaction( event );
+
+		// invalidate `sitePlans` (/sites/:site/plans)
+		this.props.clearSitePlans( this.props.selectedSite.ID );
+
+		// Refresh all sites (/sites) to have an updated `site.plan` value.
+		Dispatcher.handleViewAction( {
+			type: 'FETCH_SITES'
+		} );
+	},
+
+	isLoading: function() {
+		return ! cartItems.hasFreeTrial( this.props.cart );
```

Comment:
Since this component mimics the `secure-payment-form` from checkout which handles a `cart` and a `transaction` and will receive updates on `cart` change, it seems logical that the component is ready once its `cart` contains the expected item. I'm going to add a comment to make it clear ;)